src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -375,125 +379,128 @@ private static final class EnumMap {
-                                cb.ifeq(next);
-                            } else if (element.label() instanceof Integer integerLabel) {
-                                Label compare = cb.newLabel();
-                                Label notNumber = cb.newLabel();
-                                cb.aload(0);
-                                cb.instanceof_(Number.class.describeConstable().get());
-                                cb.ifeq(notNumber);
-                                cb.aload(0);
-                                cb.checkcast(Number.class.describeConstable().get());
-                                cb.invokevirtual(Number.class.describeConstable().get(),
-                                                 "intValue",
-                                                 MethodType.methodType(int.class)
-                                                           .describeConstable()
-                                                           .get());
-                                cb.goto_(compare);
-                                cb.labelBinding(notNumber);
-                                cb.aload(0);
-                                cb.instanceof_(Character.class.describeConstable().get());
-                                cb.ifeq(next);
-                                cb.aload(0);
-                                cb.checkcast(Character.class.describeConstable().get());
-                                cb.invokevirtual(Character.class.describeConstable().get(),
-                                                 "charValue",
-                                                 MethodType.methodType(char.class)
-                                                           .describeConstable()
-                                                           .get());
-                                cb.labelBinding(compare);
-                                cb.ldc(integerLabel);
-                                cb.if_icmpne(next);
-                            } else {
-                                throw new InternalError("Unsupported label type: " + element.label().getClass());
-                            }
-                            cb.constantInstruction(idx);
-                            cb.ireturn();
+                        lastLabel = currentLabel;
+                        cases.add(new Element(target, next, currentLabel));
+                        switchCases.add(SwitchCase.of(idx, target));
+                    }
+                    cases = cases.reversed();
+                    switchCases = switchCases.reversed();
+                    cb.tableswitch(0, labels.length - 1, dflt, switchCases);
+                    for (int idx = 0; idx < cases.size(); idx++) {
+                        Element element = cases.get(idx);
+                        Label next = element.next();
+                        cb.labelBinding(element.target());
+                        if (element.caseLabel() instanceof Class<?> classLabel &&
```

Comment:
I think you can do a if else of isPresent inside instanceof Class<?> to avoid to reapeat the instanceof and store `classLabel.describeConstable()` into a local variable.

Reply:
Done, thanks.